daemons/dss-scalability-test/app.py

```diff
@@ -0,0 +1,330 @@
+            'duration': current_time() - start_time - WAIT_CHECKOUT * 1000,
+            'success_count': success_count,
+            'fail_count': fail_count,
+            'created_on': datetime.datetime.now().isoformat(),
+            'expiration_ttl': expiration_ttl
+        }
+    )
+
+
+@app.sns_topic_subscriber("dss-scalability-test-run-" + os.environ["DSS_DEPLOYMENT_STAGE"])
+def launch_test_run(event, context):
+    msg = json.loads(event["Records"][0]["Sns"]["Message"])
+    table = dynamodb.Table('scalability_test_run')
+    table.put_item(
+        Item={
+            'run_id': msg["run_id"],
+            'executions': 0,
+            'succeeded_count': 0,
+            'failed_count': 0,
+            'average_duration': 0,
+            'created_on': datetime.datetime.now().isoformat()
+        }
+    )
+
+
+@app.sns_topic_subscriber("dss-scalability-test-" + os.environ["DSS_DEPLOYMENT_STAGE"])
+def launch_exec(event, context):
+    msg = json.loads(event["Records"][0]["Sns"]["Message"])
+    run_id = msg["run_id"]
+    execution_id = msg["execution_id"]
+    nextBatch = roundTime()
+    test_input = {
+        "execution_id": execution_id,
+        "test_run_id": run_id,
+        "batch": nextBatch.isoformat() + 'Z'
+    }
+    stepfunctions.step_functions_invoke("dss-scalability-test-{stage}", execution_id, test_input)
+
+
+@app.dynamodb_stream_handler(table_name="scalability_test", batch_size=5)
+def handle_dynamodb_stream(event, context):
+    success_count = 0
+    failure_count = 0
+    records = 0
+    duration_sum = 0
+    run_id = None
+    for record in event['Records']:
+        if record['eventName'] == 'INSERT':
+            run_id = record['dynamodb']['NewImage']['run_id']['S']
+            success_count_rec = int(record['dynamodb']['NewImage']['success_count']['N'])
+            fail_count_rec = int(record['dynamodb']['NewImage']['fail_count']['N'])
+
+            duration = record['dynamodb']['NewImage']['duration']['N']
+            duration_sum += Decimal(duration)
+            records += 1
+            success_count += success_count_rec
+            failure_count += fail_count_rec
+    print(f"success_count_rec: {success_count}")
+    if records > 0:
+        table = dynamodb.Table('scalability_test_run')
+        run_entry_pk = {'run_id': run_id}
+        run_entry = table.get_item(Key=run_entry_pk)
+        if run_entry.get('Item'):
+            old_avg_duration = run_entry['Item']['average_duration']
+            old_count = run_entry['Item']['succeeded_count'] + run_entry['Item']['failed_count']
+
+            table.update_item(
+                Key=run_entry_pk,
+                UpdateExpression='SET '
+                                 'succeeded_count = succeeded_count + :success_count, '
+                                 'failed_count = failed_count + :failure_count, '
+                                 'executions = executions + :records, '
+                                 'average_duration = :new_average',
```

Comment:
It seems like you should use ADD instead of SET here.

Reply:
In this case the attributes already exists so using SET is more appropriate.